idaes/generic_models/unit_models/heat_exchanger_lc.py

```diff
@@ -26,6 +27,22 @@
                              doc="0D heat exchanger for transient simulations")
 class HeatExchangerLumpedCapacitanceData(HeatExchangerData):
 
+    CONFIG = HeatExchangerData.CONFIG(implicit=True)
+
+    CONFIG.declare(
+        "dynamic_heat_balance",
+        ConfigValue(
+            default=True,

```

Comment:
Two comments here:

1. Do we want to allow this to take a default value based on the flowsheet, in the same way the `dynamic` flag does? In this case, we should make the default `useDefault` and the domain `DefaultBool` (see https://github.com/IDAES/idaes-pse/blob/main/idaes/core/unit_model.py).
2. I think it would be helpful to add validation of this config argument in the code (and it would be necessary if we do suggestion 1). E.g. we should check for `dynamic_heat_balance=True` in flowsheets where `dynamic=False` and raise an exception of our own to inform users what happened. Otherwise, they will eventually see a `DAE_Error`, but that is likely not going to help them understand what they did wrong and how to fix it.

Reply:
Agreed. `useDefault` makes sense here. Regarding point 2., I found this: https://github.com/IDAES/idaes-pse/blob/main/idaes/core/util/exceptions.py, which seems more appropriate than `DAE_Error`